Basics/Git.md

```diff
@@ -4,50 +4,130 @@
+
+The main branches in this flow are:
+* **master**
+  * It is a main branch
+  * anything in the master branch is deployable
+  * is stable and it is always, always safe to deploy from it or create new branches off of it
+  * feature and hotfix branches are merged into it only after pull request review is done
+* **release**
+  * used for parallel development (e.g. multiple versions of the app when version 1.0.0 is developed on master, version 1.1.0 can be a next ***release/1.1.0/*** or for developing multiple features that should or shouldn’t be in the same version: ***release/separate-cool-feature***)
+  * same rules as for the master branch
+* **feature**
+  * used for developing new features / adding new functionality to the app
+  * branches off of master/release
+  * has prefix ***feature/***, e.g. feature/123-add-followers-button
+* **fix**
+  * used for fixing non-critical production bugs or bugs occurred in development phase
+  * has prefix ***fix/***, e.g. fix/123-wrong-cards-sorting
+* **refactor**
+  * used for codebase/project refactoring
+  * Has prefix ***refactor/***, e.g. refactor/123-alert-and-action-sheet
+* **hotfix**
+  * used for fixing critical bugs in production
+  * may be branched off from the corresponding tag and create PR to the ***master*** or to the latest ***release*** branch if one exists
+  * has prefix ***hotfix/***, e.g. hotfix/1337-recipe-details-crash
+  * Example usage: there is a bug on app version 1.2.0. In that case you checkout to a commit tagged with v1.2.0. You create release/1.2.1 branch from that tag, and you create hotfix/whatever from that branch where you implement the fix and create PR back to the release/1.2.1 branch
+ 
+#### Pull requests
+Each pull request must have an automatic check - running tests or just building the app if your project doesn’t have tests. Setup on GitHub is described in general rules of this handbook, while on how to set up Bitrise Checks on projects you have [Android][7] and [iOS][8] handbook chapters.
+
+#### Deployment
+Deployment is done via tags and Bitrise CI/CD. Here are the chapters on setting up Bitrise on your project: [Android][7] and [iOS][9].
+
+Sync with another platform as much as possible. Few stuff that should be defined and synced:
+* Environment names - staging, production, uat, beta… whatever is convention on project, but should be the same. Exception probably will be iOS with its production and app store environments due to technical limitations. If using TryOutApps, please sync environment names.
+* Build number/code version - Must be numbers only value. For build number you can use commit count (could be fetched with `git rev-list HEAD --count` from deploy script and bitrise has global variable `$GIT_CLONE_COMMIT_COUNT` that could be used) or manually auto increment number value (i.e. before creating a new build, last build number should be fetched and increment by 1).
```

Comment:
We have discussed that preferred way is to use tag count, but if this is not possible, you can use commit count or manual build number entering.

Can you please change this paragraph for the build number. Bitrise part and deploy script are already updated with tag count.

Additionally, please add mention for deploy script somewhere :)